cloud/gcp/deploy/website.go

```diff
@@ -18,13 +18,349 @@ package deploy
+		return nil
+	}
+
+	return fmt.Errorf("%s is not a valid subdomain of %s", subdomain, domain)
+}
+
+// Deploy a cloud CDN entrypoint
+func (a *NitricGcpPulumiProvider) deployEntrypoint(ctx *pulumi.Context) error {
+	if a.GcpConfig.CdnDomain.ZoneName == "" {
+		return fmt.Errorf("a valid DNS zone is required to deploy websites to GCP")
+	}
+
+	if a.GcpConfig.CdnDomain.DomainName == "" {
+		return fmt.Errorf("a valid domain name is required to deploy websites to GCP")
+	}
+
+	pathRules := compute.URLMapPathMatcherPathRuleArray{}
+
+	// Add deployed API gateways to the URLMap
+	for apiName, api := range a.ApiGateways {
+		neg, err := compute.NewRegionNetworkEndpointGroup(ctx, fmt.Sprintf("%s-apigw-neg", apiName), &compute.RegionNetworkEndpointGroupArgs{
+			NetworkEndpointType: pulumi.String("SERVERLESS"),
+			Region:              api.Region,
+			ServerlessDeployment: compute.RegionNetworkEndpointGroupServerlessDeploymentArgs{
+				Platform: pulumi.String("apigateway.googleapis.com"),
+				Resource: api.GatewayId,
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		bs, err := compute.NewBackendService(ctx, fmt.Sprintf("%s-apigw-bs", apiName), &compute.BackendServiceArgs{
+			Backends: compute.BackendServiceBackendArray{
+				compute.BackendServiceBackendArgs{
+					Group: neg.SelfLink,
+				},
+			},
+			Protocol: pulumi.String("HTTPS"),
+		}, nil)
+		if err != nil {
+			return err
+		}
+
+		pr := compute.URLMapPathMatcherPathRuleArgs{
+			Service: bs.ID(),
+			Paths:   pulumi.StringArray{pulumi.Sprintf("/apis/%s/*", apiName)},
+			RouteAction: compute.URLMapPathMatcherPathRuleRouteActionArgs{
+				UrlRewrite: compute.URLMapPathMatcherPathRuleRouteActionUrlRewriteArgs{
+					PathPrefixRewrite: pulumi.String("/"),
+					HostRewrite:       api.DefaultHostname,
+				},
+			},
+		}
+
+		pathRules = append(pathRules, pr)
+	}
+
+	cdnPolicy := compute.BackendBucketCdnPolicyArgs{}
+
+	if a.GcpConfig.CdnDomain.ClientTtl != nil {
+		cdnPolicy.ClientTtl = pulumi.Int(*a.GcpConfig.CdnDomain.ClientTtl)
+	}
+
+	if a.GcpConfig.CdnDomain.DefaultTtl != nil {
+		cdnPolicy.DefaultTtl = pulumi.Int(*a.GcpConfig.CdnDomain.DefaultTtl)
+	}
+
+	var defaultService pulumi.StringOutput
+	for sitePath, siteBucket := range a.WebsiteBuckets {
+		normalizedName := strings.Replace(sitePath, "/", "", -1)
+		if normalizedName == "" {
+			normalizedName = "default"
+		}
+
+		backend, err := compute.NewBackendBucket(ctx, fmt.Sprintf("%s-site-bucket", normalizedName), &compute.BackendBucketArgs{
+			BucketName:      siteBucket.Name,
+			EnableCdn:       pulumi.Bool(true),
+			CompressionMode: pulumi.String("AUTOMATIC"),
+			CdnPolicy:       cdnPolicy,
+		})
+		if err != nil {
+			return err
+		}
+
+		if sitePath == "/" {
+			defaultService = backend.SelfLink
+		} else {
+			pr := compute.URLMapPathMatcherPathRuleArgs{
+				Service: backend.ID(),
+				Paths:   pulumi.StringArray{pulumi.String(filepath.Join("/", sitePath, "./*"))},
```

Comment:
I am getting [0]: "\\vue\\*" for this, which is invalid. Needs to be cleaned to work on windows.

Reply:
Diagnostics:
  gcp:compute:URLMap (https-site-url-map):
    error: 1 error occurred:
        * Error creating UrlMap: googleapi: Error 400: Invalid value for field 'resource': '{  "name": "https-site-url-map-4c2cd12",  "hostRules": [{    "hosts":
["*"],    "pathMatcher": "all-...'. Invalid path pattern, invalid

  pulumi:pulumi:Stack (test-web-test-web-gcpp):
    error: update failed

  gcp:compute:URLMap (https-site-url-map
):
    error:   sdk-v2/provider2.go:509: sdk.helper_schema: Error creating UrlMap: googleapi: Error 400: Invalid value for field 'resource': '{  "name": "https-site
-
url-map-4c2cd12",  "hostRules": [{    "hosts": ["*"],    "pathMatcher": "all-...'. Invalid path pattern, invalid: provider=google-beta@8.4.1

Reply:
See suggested fix below